design_drafts/ros2_control_documentation.rst

```diff
@@ -0,0 +1,273 @@
+The method can access the latest hardware states and enables the controller to write the hardware's command interfaces.
+
+User Interfaces
+---------------
+Users interact with the ros2_control framework using `Controller Manager`_'s services.
+Those can be used directly or through the "cli"-interface (base command: ``ros2 control``).
+The "cli"-interface is more user-friendly for direct interaction, i.e., outside of a node.
+
+For a list of services and their definitions, check the ``srv`` folder in the `controller_manager_msgs`_ package.
+
+For the description of the "cli"-interface, see the ``README.md`` file of the `ros2controlcli`_ package.
+
+
+Hardware Components
+===================
+The *hardware components* realize communication to physical hardware and represent its abstraction in the ros2_control framework.
+The components have to be exported as plugins using ``pluginlib``-library.
+The `Resource Manager`_ dynamically loads those plugins and manages their lifecycle.
+
+There are three basic types of components:
+
+Actuators
+  Simple (1 DOF) robotic hardware like motors, valves, and similar. 
+  An actuator implementation is related to only one joint.
+  This component type has reading and writing capabilities. Reading is not mandatory if not possible (e.g., DC motor control with Arduino board).
+  The actuator type can also be used with a multi-DOF robot if its hardware enables modular design, e.g., CAN-communication with each motor independently.
+  
+Sensors
+  Robotic hardware is used for sensing its environment.
+  A sensor component is related to a joint (e.g., encoder) or a link (e.g., force-torque sensor).
+  This component type has only reading capabilities.
+  
+System
+  Complex (multi-DOF) robotic hardware like industrial robots.
+  The main difference between the *Actuator* component is the possibility to use complex transmissions like needed for humanoid robot's hands.
+  This component has reading and writing capabilities.
+  It is used when the is only one logical communication channel to the hardware (e.g., KUKA-RSI).  
+
+A detailed explanation of hardware components is given in the `Hardware Access through Controllers design document`_.
+
+Hardware Description in URDF
+----------------------------
+The ros2_control framework uses the ``<ros2_control>``-tag in the robot's URDF file to describe its components, i.e., the hardware setup.
+The chosen structure enables tracking together multiple `xacro`-macros into one without any changes. 
+The example hereunder shows a position-controlled robot with 2-DOF (RRBot), an external 1-DOF force-torque sensor, and an externally controlled 1-DOF parallel gripper as its end-effector.
+For more examples and detailed explanations, check `ros-controls/ros2_control_demos`_ repository and `ROS2 Control Components URDF Examples design document`_.
+
+.. code:: xml
+
+<ros2_control name="RRBotSystemPositionOnly" type="system">
```

Comment:
the "rich view" mode doesn't seem to display this section properly

Reply:
Afaik all lines which should be part of the block need to be indented.

Reply:
@gavanderhoorn thanks that did it!